RadixWallet/Features/ShieldSetup/SelectFactorSources/Children/SelectFactorSources+View.swift

```diff
@@ -0,0 +1,134 @@
+import ComposableArchitecture
+import SwiftUI
+
+// TODO: move to Sargon
+extension FactorSourceKind {
+	var displayOrder: Int {
+		switch self {
+		case .device: 0
+		case .arculusCard: 1
+		case .ledgerHqHardwareWallet: 2
+		case .password: 3
+		case .offDeviceMnemonic: 4
+		case .trustedContact: 5
+		case .securityQuestions: 6
+		}
+	}
+}
+
+extension SelectFactorSources.State {
+	var sortedFactorSources: [FactorSource] {
```

Comment:
with `displayOrder` in sargon (as per your comment) we will have the main logic of this sorting in Sargon, which is great. But I wonder if we should not try to take it any further! This is kind of still having logic in hosts. I think better if Sargon provided this **whole** `Reducer.State` with its (view)state! i.e. (pseudocode)

```swift
struct ParentReduer {
    func reduce(internalAction: InternalAction) {
       switch internalAction {
          case .selectFactorSourcePressed:
            let sargonPreparedChildState = SargonOS.viewModel(for: .selectFactorSources)
            let childReducerState = SelectFactorSources(state: sargonPreparedChildState)
             state.destination = .selectFactorSources(state: childReducerState)
      }
    }
}
```

Reply:
merits a bigger decision and Ghenadies presence. But I think we should be really vary of putting **any** logic whatsoever in hosts. And this sorting has a liiitle bit of implicit logic. Since it "decides" that it should use `<` and `$0.factorSourceKind.displayOrder` - which is much more logic than `SargonOS.sortFactorsForSelection(factors: factors)` or as above, letting Sargon build the "ViewModel" (state in `Reducer.State` I guess).

Reply:
`SargonOS.sortFactorsForSelection(factors: factors)` seems fair enough for now

Reply:
@danvleju-rdx I ofc meant `FactorSources` not `Factors` (since `Factors` is forbidden! I was just lazy...) 
probably should not live in `SargonOS` namespace. Probably a method on `SecurityShieldBuilder`
```swift
SecurityShieldBuilder().sortFactorsSourcesForSelection(factors)
```


```rust
#[uniffi::export]
impl SecurityShieldBuilder {
    fn sort_factor_sources_for_selection(&self, factor_sourcess: [FactorSources]) ->  { .. }
}
```

That wat you have a more suitable namespace! no need to polute SargonOS with it. and also that way you dont need to have.

Or see if you can `Uniffi` export a ("static") function in UniFFI:

```rust
#[uniffi::export]
impl SecurityShieldBuilder {
// Note lacke of `&self`....
    fn sort_factor_sources_for_selection(factor_sourcess: [FactorSources]) ->  { .. }
}
```

Im not sure if that works. If so you have two solutions:
1. uniffi::export a global freestanding function
2. use the method above, but you might do an `extension SecurityShieldBuilder` in **Swift** Sargon and `public static func` it and use it on a new instance of `SecurityShieldBuilder`. I think making it a `static func` under `SecurityShieldBuilder` is what we want, to make it clear that it is **not a method which mutates the internal state of the `SecurityShieldBuilder`**! Rather just a utility function which we put under the `SecurityShieldBuilder`'s namespace!

Reply:
agree with the final conclusions: a static exported method on `SecurityShieldBuilder` seems the best choice for now